### DISCUSSION TITLE
Pre fit and post fit, CLs plots

### DISCUSSION THREAD

**Discussion by ssaha1234** [scikit-hep/pyhf#1769] [Q&A] [answered]
Dear pyhf community, 
   I am making prefit and postfit plots for my analysis (similar to what we have in HistFitter), as well as limit setting plots. 
Are there some examples scripts to start off with ?

Thank you very much !
Best, 
Shreya

**Comment by alexander-held** [comment, marked as answer]
Hi Shreya, `pyhf` provides a function to draw the Cls distribution via [`contrib.viz.brazil`](https://pyhf.readthedocs.io/en/v0.6.3/_generated/pyhf.contrib.viz.brazil.html).

 [`cabinetry`](https://cabinetry.readthedocs.io/) provides functionality to calculate the required information for plots and to draw figures, without having to implement everything from scratch. For pre-/post-fit plots, the relevant modules of `cabinetry` are [`model_utils`](https://cabinetry.readthedocs.io/en/latest/api.html#module-cabinetry.model_utils) (includes calculation of yields & uncertainties) and [`visualize`](https://cabinetry.readthedocs.io/en/latest/api.html#module-cabinetry.visualize) (plotting). `visualize` also can draw a CLs distribution. You can find some examples of the usage in these two notebooks: [notebook 1](https://github.com/cabinetry/cabinetry-tutorials/blob/08a0363db0585d12bdcbc9600c9e139351142e8b/example.ipynb), [notebook 2](https://github.com/cabinetry/cabinetry-tutorials/blob/08a0363db0585d12bdcbc9600c9e139351142e8b/HEPData_workspace.ipynb) (from the [`cabinetry-tutorials`](https://github.com/cabinetry/cabinetry-tutorials) repository). For `cabinetry`, it might also be interesting to note that there is a significant performance improvement for the post-fit uncertainty calculation in preparation (https://github.com/scikit-hep/cabinetry/issues/315).

**Reply by matthewfeickert** [reply]
There are some additional plotting examples (just with `pyhf`) in the `pyhf` tutorial: https://pyhf.github.io/pyhf-tutorial/

**Reply by ssaha1234** [reply]
Thanks @matthewfeickert  for sharing the additional examples :) 
Best, 
Shreya

**Comment by ssaha1234** [comment]
Hi @alexander-held, 
    Thank you very much for the prompt reply and all the links :)) I will read through it and implement it, starting from the examples ! 
Thanks !
Best,
Shreya
